spec/migrations/migrate_admin_tax_amounts_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require_relative '../../db/migrate/20210617203927_migrate_admin_tax_amounts'
+
+describe MigrateAdminTaxAmounts do
+  subject { MigrateAdminTaxAmounts.new }
+
+  let(:tax_category10) { create(:tax_category) }
+  let(:tax_category50) { create(:tax_category) }
+  let!(:tax_rate10) { create(:tax_rate, amount: 0.1, tax_category: tax_category10) }
+  let!(:tax_rate50) { create(:tax_rate, amount: 0.5, tax_category: tax_category50) }
+  let(:adjustment10) { create(:adjustment, amount: 100, included_tax: 10) }
+  let(:adjustment50) { create(:adjustment, amount: 100, included_tax: 50) }
+
+  describe '#migrate_admin_taxes!' do
+    context "when the adjustment has no tax" do
+      let!(:adjustment_without_tax) { create(:adjustment, included_tax: 0) }
+
+      it "doesn't move the tax to an adjustment" do
+        expect(Spree::Adjustment).to_not receive(:create!)
+
+        subject.migrate_admin_taxes!
```

Comment:
This requires knowledge of the internal working (using `create!`) which may change in the future (eg `new(..).save!`. I've seen this solution in the past:
```suggestion
        expect { subject.migrate_admin_taxes! }.to_not change {
          Spree::Adjustment.count
        }
```